### PR TITLE
Various entity and collision related mappings

### PR DESCRIPTION
**Pull request by Siphalor**

### CODE REVIEW

File: mappings/net/minecraft/entity/Entity.mapping [hunk 1]
```diff
@@ -25,6 +26,8 @@ CLASS ahc net/minecraft/entity/Entity
 	FIELD aH NO_GRAVITY Lps;
 	FIELD aI invulnerable Z
 	FIELD aJ scoreboardTags Ljava/util/Set;
+	FIELD aL pistonMovement [D
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/Entity.mapping]: pistonMovementDelta?

File: mappings/net/minecraft/entity/Entity.mapping [hunk 2]
```diff
@@ -3,7 +3,8 @@ CLASS ahc net/minecraft/entity/Entity
 	FIELD B verticalCollision Z
 	FIELD C collided Z
 	FIELD D velocityModified Z
-	FIELD E movementMultiplierSet Z
+	FIELD E applyMovementMultipliers Z
+	FIELD F movementMultipliers Lcmr;
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/Entity.mapping]: No plural here

File: mappings/net/minecraft/util/SystemUtil.mapping [hunk 3]
```diff
@@ -21,6 +21,7 @@ CLASS m net/minecraft/util/SystemUtil
 	METHOD a createTranslationKey (Ljava/lang/String;Lqd;)Ljava/lang/String;
 		ARG 0 type
 		ARG 1 id
+	METHOD a tryConsumeOrRun (Ljava/util/Optional;Ljava/util/function/Consumer;Ljava/lang/Runnable;)Ljava/util/Optional;
```
Comment (asiekierka) [inline mappings/net/minecraft/util/SystemUtil.mapping]: I do not like this name.

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/SystemUtil.mapping (hunk 3 above)
Comment (Siphalor) [reply mappings/net/minecraft/util/SystemUtil.mapping]: I don't like it either but I didn't find a better one.

The method does the following:
If the `Optional` is present it executes the `Consumer` with it.
If not it runs the `Runnable`.

Do you have any suggestions?

Comment (asiekierka) [reply mappings/net/minecraft/util/SystemUtil.mapping]: Just drop it for now.

Comment (Siphalor) [reply mappings/net/minecraft/util/SystemUtil.mapping]: Did it

File: mappings/net/minecraft/entity/Entity.mapping [hunk 4]
```diff
@@ -52,6 +55,7 @@ CLASS ahc net/minecraft/entity/Entity
 	FIELD g type Lahg;
 	FIELD h LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD j ridingCooldown I
+	FIELD k isTeleporting Z
```
Comment (liach) [inline mappings/net/minecraft/entity/Entity.mapping:58]: `teleporting`

File: mappings/net/minecraft/entity/Entity.mapping [hunk 5]
```diff
@@ -179,6 +184,8 @@ CLASS ahc net/minecraft/entity/Entity
 	METHOD ab setOnFireFromLava ()V
 	METHOD ac extinguish ()V
 	METHOD ad destroy ()V
+	METHOD af centerAtBoundingBox ()V
```
Comment (liach) [inline mappings/net/minecraft/entity/Entity.mapping:187]: Hmm, what exactly does this do to the entity?

File: mappings/net/minecraft/entity/EntityType.mapping [hunk 6]
```diff
@@ -124,7 +131,11 @@ CLASS ahg net/minecraft/entity/EntityType
 		ARG 4 summonable
 		ARG 5 dataFixerType
 	METHOD a isSaveable ()Z
+	METHOD a createSimpleBoundingBox (DDD)Lcmm;
 	METHOD a createInstanceFromId (ILbcl;)Lahc;
+	METHOD a spawnEntityStack (Lahc;Lbcm;)V
```
Comment (liach) [inline mappings/net/minecraft/entity/EntityType.mapping:136]: Uh, what is an entity stack?

File: mappings/net/minecraft/world/ViewableWorld.mapping [hunk 7]
```diff
@@ -40,7 +54,15 @@ CLASS bcp net/minecraft/world/ViewableWorld
...
 	METHOD b isChunkLoaded (II)Z
 	METHOD b getClosestSurvivalPlayer (Lahc;D)Larj;
+	METHOD b getCollidingBoundingBoxesForEntity (Lahc;Lcmm;)Ljava/util/stream/Stream;
+		ARG 1 entity
+		ARG 2 entityBoundingBox
+		ARG 3 entityVertical
 	METHOD c getAmbientDarkness ()I
+	METHOD c isEntityColliding (Lahc;Lcmm;)Z
+		ARG 1 entity
+		ARG 2 entiryBoundingBox
```
Comment (YanisBft) [inline mappings/net/minecraft/world/ViewableWorld.mapping:64]: Typo here

File: mappings/net/minecraft/entity/EntityType.mapping (hunk 6 above)
Comment (liach) [reply mappings/net/minecraft/entity/EntityType.mapping:136]: How about `spawnEntityWithPassengers`

File: mappings/net/minecraft/util/SystemUtil.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/util/SystemUtil.mapping]: FYI in java 9 optional has `ifPresentOrElse`.

File: mappings/net/minecraft/entity/EntityType.mapping [hunk 8]
```diff
@@ -124,7 +131,11 @@ CLASS ahg net/minecraft/entity/EntityType
...
+		ARG 0 entity
+		ARG 1 iWorld
```
Comment (Gegy) [inline mappings/net/minecraft/entity/EntityType.mapping:138]: `world`

File: mappings/net/minecraft/world/ViewableWorld.mapping [hunk 9]
```diff
@@ -27,6 +27,20 @@ CLASS bcp net/minecraft/world/ViewableWorld
 		ARG 3 status
 		ARG 4 create
 	METHOD a getClosestPlayer (Lahc;D)Larj;
+	METHOD a isEntityColliding (Lahc;Lcmm;Ljava/util/Set;)Z
+		ARG 1 entity
+		ARG 2 entityBoundingBox
+		ARG 3 other
+		ARG 4 entityVertical
+	METHOD a getCollidingBoundingBoxesForEntity (Lahc;Lcnl;Lcnl;Ljava/util/Set;Lcmw;)Ljava/util/stream/Stream;
```
Comment (Gegy) [inline mappings/net/minecraft/world/ViewableWorld.mapping:35]: These are quite long but can't really think of better alternatives. Thoughts?

File: mappings/net/minecraft/util/math/BoundingBox.mapping [hunk 10]
```diff
@@ -14,30 +14,94 @@ CLASS cmm net/minecraft/util/math/BoundingBox
...
+		ARG 4 minY
+		ARG 6 minZ
+		ARG 8 maxX
+		ARG 10 maxY
+		ARG 12 maxZ
 	METHOD a offset (Lcmr;)Lcmm;
+		ARG 1 vec3d
 	METHOD a intersects (Lcmr;Lcmr;)Z
 		ARG 1 a
+		ARG 2 vec3d2
```
Comment (modmuss50) [inline mappings/net/minecraft/util/math/BoundingBox.mapping:48]: `b` ? this doesnt seem right

**Pull request closed without merging**